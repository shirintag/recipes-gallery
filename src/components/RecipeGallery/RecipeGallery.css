.recipes-gallery {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;
  text-align: center;
}
