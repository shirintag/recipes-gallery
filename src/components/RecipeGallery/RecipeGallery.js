import React from "react";
// import recipes from "../../../recipes.json";
import RecipeCard from "components/RecipeCard/RecipeCard";
import "./RecipeGallery.css";

const RecipesGallery = ({ recipes }) => {
  const products = recipes.map((recipe) => (
    <RecipeCard {...recipe} key={recipe.id} />
  ));
  return <div className="recipes-gallery">{products}</div>;
};

export default RecipesGallery;
