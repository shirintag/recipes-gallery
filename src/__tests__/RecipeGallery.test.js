import React from "react";
import { render } from "@testing-library/react";
import RecipeData from "../../recipes.json";
import RecipeGallery from "../components/RecipeGallery/RecipeGallery";

describe("RecipeGallery", () => {
  it("should render", () => {
    const { container } = render(<RecipeGallery recipes={RecipeData} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
