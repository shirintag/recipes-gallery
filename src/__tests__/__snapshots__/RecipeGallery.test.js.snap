// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeGallery should render 1`] = `
<div
  class="recipes-gallery"
>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Crispy Fish Goujons "
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Crispy Fish Goujons 
      </div>
      <div
        class="f-subtitle"
      >
        with Sweet Potato Wedges and Minted Snap Peas
      </div>
      <span>
        Calories: 
        516 kcal
      </span>
      <span>
        Time: 
        35M
      </span>
    </div>
  </div>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Moroccan Skirt Steak "
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Moroccan Skirt Steak 
      </div>
      <div
        class="f-subtitle"
      >
        with Spinach and Lemon Couscous
      </div>
      <span>
        Calories: 
        397 kcal
      </span>
      <span>
        Time: 
        30M
      </span>
    </div>
  </div>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Simple Sumptuous Sea Bream"
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Simple Sumptuous Sea Bream
      </div>
      <div
        class="f-subtitle"
      >
        with Tomato Concasse and Crispy Potatoes
      </div>
      <span>
        Calories: 
        458 kcal
      </span>
      <span>
        Time: 
        35M
      </span>
    </div>
  </div>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Mozzarella and Pesto Roulades"
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Mozzarella and Pesto Roulades
      </div>
      <div
        class="f-subtitle"
      >
        with Roasted Rocket Potatoes
      </div>
      <span>
        Calories: 
        717 kcal
      </span>
      <span>
        Time: 
        35M
      </span>
    </div>
  </div>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Mexican Tortilla Stack"
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Mexican Tortilla Stack
      </div>
      <div
        class="f-subtitle"
      >
        with Homemade Guacamole and Black Bean Salsa
      </div>
      <span>
        Calories: 
        751 kcal
      </span>
      <span>
        Time: 
        35M
      </span>
    </div>
  </div>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Roasted Rosemary Butternut Squash "
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Roasted Rosemary Butternut Squash 
      </div>
      <div
        class="f-subtitle"
      >
        with Wild Rice, Feta and Pine Nuts
      </div>
      <span>
        Calories: 
        689 kcal
      </span>
      <span>
        Time: 
        40M
      </span>
    </div>
  </div>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Gnocchi Allo Genovese"
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Gnocchi Allo Genovese
      </div>
      <div
        class="f-subtitle"
      >
        with Toasted Pine Nuts and Tenderstem Broccoli
      </div>
      <span>
        Calories: 
        --
      </span>
      <span>
        Time: 
        25M
      </span>
    </div>
  </div>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Herbed Italian Polpetti"
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Herbed Italian Polpetti
      </div>
      <div
        class="f-subtitle"
      >
        with Creamy Parmesan Polenta
      </div>
      <span>
        Calories: 
        790 kcal
      </span>
      <span>
        Time: 
        35M
      </span>
    </div>
  </div>
  <div
    class="recipe-card"
  >
    <div
      class="recipe-heart"
    >
      <div>
        <div
          class="heart"
        />
      </div>
    </div>
    <div
      class="recipe-image-container"
    >
      <img
        alt="Genki Yakitori with Crispy Red Onions"
        class="recipe-image"
        src="https://i2.wp.com/www.downshiftology.com/wp-content/uploads/2018/12/Shakshuka-19.jpg"
      />
    </div>
    <div
      class="recipe-data f-subcopy"
    >
      <div
        class="star-rating"
      >
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
        <div
          class="star-container"
        >
          <div
            class="star"
          />
        </div>
      </div>
      <div
        class="f-title"
      >
        Genki Yakitori with Crispy Red Onions
      </div>
      <div
        class="f-subtitle"
      >
        and Sweet and Sour Noodles
      </div>
      <span>
        Calories: 
        556 kcal
      </span>
      <span>
        Time: 
        40M
      </span>
    </div>
  </div>
</div>
`;
